src/Producer.php

```diff
@@ -118,8 +119,15 @@ public function send($data = null, $routingKey = null)
 
         $message = new AMQPMessage($data, ['delivery_mode' => $this->deliveryMode]);
 
+        /** @var LifecycleManager $lifecycle */
+        $lifecycle = app('vinelab.bowler.lifecycle');
+
+        $message = $lifecycle->triggerBeforePublish($message, $this->exchangeName, $routingKey);
```

Comment:
failures here would stop the message from being published. Also better be failsafe.

Reply:
Executed callback are now [wrapped](https://github.com/Vinelab/bowler/pull/49/files#diff-eb2ac16609a9bd82b169a141bf74c71cR156) in a try-catch to ensure the execution proceeds in case of an error.